### ISSUE TITLE
Firmware update fails on Nitrokey 3A NFC

### ISSUE THREAD

**Issue by leitmedium** [Nitrokey/pynitrokey#270]
Hello,

the update of the firmware on a Nitrokey 3A NFC fails. Session on macOS looks like:

```
➜  ~ nitropy nk3 update
Command line tool to interact with Nitrokey devices 0.4.25
Current firmware version:  v1.0.4
Latest firmware version:   v1.2.0
Do you want to download the firmware version v1.2.0? [Y/n]: y

Please do not remove the Nitrokey 3 or insert any other Nitrokey 3 devices during the update. Doing so may damage the Nitrokey 3.
Do you want to perform the firmware update now? [y/N]: y

Please press the touch button to reboot the device into bootloader mode ...

Bootloader mode enabled. Please repeat this command to apply the update.
Aborted!
➜  ~ nitropy nk3 update
Command line tool to interact with Nitrokey devices 0.4.25
Current firmware version:  [unknown]
Latest firmware version:   v1.2.0
Do you want to download the firmware version v1.2.0? [Y/n]:

Please do not remove the Nitrokey 3 or insert any other Nitrokey 3 devices during the update. Doing so may damage the Nitrokey 3.
Do you want to perform the firmware update now? [y/N]: y
Critical error:
An unhandled exception occurred
	Exception encountered: UnboundLocalError("local variable 'bootloader' referenced before assignment")

--------------------------------------------------------------------------------
Critical error occurred, exiting now
Unexpected? Is this a bug? Would you like to get support/help?
- You can report issues at: https://support.nitrokey.com/
- Writing an e-mail to support@nitrokey.com is also possible
- Please attach the log: '/var/folders/f_/hyhfl3n50_l5f2pvl5gz43qc0000gn/T/nitropy.log.ytjc2i3o' with any support/help request!
--------------------------------------------------------------------------------
```

Most important part from debug log:

```
2979      DEBUG       root print: Critical error:
2979      DEBUG       root print: An unhandled exception occurred
2979      ERROR       root local variable 'bootloader' referenced before assignment
Traceback (most recent call last):
  File "/Users/caspar/.local/pipx/venvs/pynitrokey/lib/python3.10/site-packages/pynitrokey/cli/__init__.py", line 111, in main
    nitropy()
  File "/Users/caspar/.local/pipx/venvs/pynitrokey/lib/python3.10/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/caspar/.local/pipx/venvs/pynitrokey/lib/python3.10/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/caspar/.local/pipx/venvs/pynitrokey/lib/python3.10/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/caspar/.local/pipx/venvs/pynitrokey/lib/python3.10/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/caspar/.local/pipx/venvs/pynitrokey/lib/python3.10/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/caspar/.local/pipx/venvs/pynitrokey/lib/python3.10/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/caspar/.local/pipx/venvs/pynitrokey/lib/python3.10/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/Users/caspar/.local/pipx/venvs/pynitrokey/lib/python3.10/site-packages/pynitrokey/cli/nk3/__init__.py", line 372, in update
    update_version = update(ctx, image, variant)
  File "/Users/caspar/.local/pipx/venvs/pynitrokey/lib/python3.10/site-packages/pynitrokey/cli/nk3/update.py", line 84, in update
    firmware_or_release, current_version, bootloader.variant
```

**Comment by robin-nitrokey** [comment]
This bug has been fixed in pynitrokey v0.4.26.  Unfortunately, this version is not available on Arm systems yet because of a missing dependency, see https://github.com/Nitrokey/pynitrokey/issues/265.  We are currently investigating possible workarounds.  In the meantime, can you try installing pynitrokey with Rosetta (along the lines of [this guide](https://stackoverflow.com/questions/71691598/how-to-run-python-as-x86-with-rosetta2-on-arm-macos-machine)) or using a non-Arm machine?

**Comment by leitmedium** [comment]
Hi Robin, 

thanks for the reply. I successfully deployed the update with version 0.4.28 on x86.

Closing the ticket.
